conbench/entities/benchmark_result.py

```diff
@@ -598,13 +599,40 @@ def ui_mean_and_uncertainty(values: List[float], unit: str) -> str:
     return f"({mean_uncertainty_str}) {unit}"
 
 
+def validate_augment_unit_string(u: str) -> str:
+    """
+    Raise BenchmarkResultValidationError for invalid unit string.
+
+    Return augmented unit string, to be inserted into database.
+    """
+    if u == "b/s":
+        # Rewrite short variant here for a legacy client, see
+        # https://github.com/conbench/conbench/issues/1335
+        # We might need/want to do a database migration where we rewrite
+        # b/s to B/s.
+        u = "B/s"
```

Comment:
I think this means we have to do a DB migration. At first I thought "ah, then I don't do the rewrite". But then, well.. not a good state either, kicking the can down the road. I don't want to describe to users that we allow `b/s`.

Reply:
Per [this line](https://github.com/apache/arrow/blob/819b7d5ce7febb4af9ea6f69947244f46403ce53/ci/scripts/go_bench_adapt.py#L86) I'm pretty confident `B/s` is correct. But yes, we should have `@zeroshade` review the PR.